### ISSUE TITLE
Unable to find an initialized, responsive instance of Fermentrack

### ISSUE THREAD

**Issue by adrianobrum** [thorrak/fermentrack-tools#49]
The second time trying to install the Fermentrack but there are some errors that crash my installation.
Problem start after the verification if PostgreSQL is available.

> PostgreSQL is available
> 
> Traceback (most recent call last):
>   File "manage.py", line 25, in <module>
>     execute_from_command_line(sys.argv)
>   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 401, in execute_from_command_line
>     utility.execute()
>   File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 377, in execute
>     django.setup()
>   File "/usr/local/lib/python3.8/site-packages/django/__init__.py", line 24, in setup
>     apps.populate(settings.INSTALLED_APPS)
>   File "/usr/local/lib/python3.8/site-packages/django/apps/registry.py", line 114, in populate
>     app_config.import_models()
>   File "/usr/local/lib/python3.8/site-packages/django/apps/config.py", line 211, in import_models
>     self.models_module = import_module(models_module_name)
>   File "/usr/local/lib/python3.8/importlib/__init__.py", line 127, in import_module
>     return _bootstrap._gcd_import(name[level:], package, level)
>   File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
>   File "<frozen importlib._bootstrap>", line 991, in _find_and_load
>   File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
>   File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
>   File "<frozen importlib._bootstrap_external>", line 843, in exec_module
>   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>   File "/app/app/models.py", line 21, in <module>
>     from lib.ftcircus.client import CircusMgr, CircusException
>   File "/app/lib/ftcircus/client.py", line 4, in <module>
>     from circus.client import CircusClient
>   File "/usr/local/lib/python3.8/site-packages/circus/client.py", line 6, in <module>
>     import zmq
>   File "/usr/local/lib/python3.8/site-packages/zmq/__init__.py", line 50, in <module>
>     from zmq import backend
>   File "/usr/local/lib/python3.8/site-packages/zmq/backend/__init__.py", line 40, in <module>
>     reraise(*exc_info)
>   File "/usr/local/lib/python3.8/site-packages/zmq/utils/sixcerpt.py", line 34, in reraise
>     raise value
>   File "/usr/local/lib/python3.8/site-packages/zmq/backend/__init__.py", line 27, in <module>
>     _ns = select_backend(first)
>   File "/usr/local/lib/python3.8/site-packages/zmq/backend/select.py", line 28, in select_backend
>     mod = __import__(name, fromlist=public_api)
>   File "/usr/local/lib/python3.8/site-packages/zmq/backend/cython/__init__.py", line 6, in <module>
>     from . import (constants, error, message, context,
> ImportError: /usr/local/lib/python3.8/site-packages/zmq/backend/cython/error.cpython-38-arm-linux-gnueabihf.so: undefined symbol: zmq_strerror
> 
> Sentry is attempting to send 2 pending error messages
> Waiting up to 2 seconds
> Press Ctrl-C to quit
> fermentrack-tools_postgres_1 is up-to-date
> fermentrack-tools_redis_1 is up-to-date
> Creating fermentrack-tools_django_1 ... done
> Creating fermentrack-tools_nginx_1  ... done
> ::: Waiting for Fermentrack install to initialize and become responsive.
> ::: Fermentrack may take up to 3 minutes to first boot as the database is being initialized.
> ..........................................................................................missing.
> *** ERROR: Unable to find an initialized, responsive instance of Fermentrack

**Comment by thorrak** [comment]
What version of Raspberry Pi OS are you using?

**Comment by adrianobrum** [comment]
Raspberry Pi OS: Raspbian GNU/Linux 11 (bullseye)
Raspberry Pi Board: 3 B

Is this new? I can try with the Legacy Version 10 (Buster).

**Comment by thorrak** [comment]
> Raspberry Pi OS: Raspbian GNU/Linux 11 (bullseye) Raspberry Pi Board: 3 B
> 
> Is this new? I can try with the Legacy Version 10 (Buster).

It is, but I actually spent the past weekend rebuilding my RPi cluster and updating to bullseye. I’m pretty sure I know what is happening - let me poke at this!

**Comment by adrianobrum** [comment]
Great! Until there I will try with the Legacy.

**Comment by thorrak** [comment]
Let me know if that works — my current hypothesis would point at legacy failing as well.

As part of the work I did this past weekend, I built a new release of Fermentrack. Looking through the build logs it appears that one of the libraries Fermentrack uses updated itself (despite supposedly being locked to a specific, working version). If this is in fact the cause of the error it should persist regardless of the Raspberry Pi OS version and will require a rebuild on my end.

I’m waiting on a part (being delivered today!) to support performing this rebuild - I should be able to get this out tonight.

**Comment by lbussy** [comment]
Huh. Mine works. :)

**Comment by thorrak** [comment]
> Huh. Mine works. :)

When did you last install the docker containers?

**Comment by lbussy** [comment]
Docker is for suckers.

**Comment by adrianobrum** [comment]
Hi again.

First, I tried to install it on legacy but the problem is still the same.
Second, I tried to install the version without Docker and it was a success. When I tried to access the web page it seems a little broken and I was not able to add devices. I tried the pinned issue https://github.com/thorrak/fermentrack/issues/463 but is the same.

![image](https://user-images.githubusercontent.com/6775289/146266480-75fdf2b4-050b-4dc6-851d-ae734db7549f.png)

**Comment by thorrak** [comment]
Try now (with the docker version)

**Comment by thorrak** [comment]
I just pushed updated images to Docker hub - Try running that now and let me know if it works for you.

**Comment by adrianobrum** [comment]
It is perfect! Tried with Bullseye and it seems to be working well. Later I will try to flash ESP8266 and if anything fails I open a new issue.
Thank you, you saved my beer 🍻
